### PR TITLE
os: add function os.loginname(), make some error improvements 

### PR DESCRIPTION
**Pull request by bstnbuck**
This PR implements a new OS function loginname(). Furthermore it adds a better error handling in file_size() and truncate().
- add os.loginname() returns name of logged in user
- improvements for error handling in os.file_size() and os.truncate()

### CODE REVIEW

File: vlib/os/os_nix.c.v [hunk 1]
```diff
@@ -6,6 +6,7 @@ import strings
 #include <unistd.h>
 #include <fcntl.h>
 #include <sys/utsname.h>
+#include <utime.h>
```
Comment (bstnbuck) [inline vlib/os/os_nix.c.v:10]: This include causes the CI macos-cross failure. Don't know how to fix it...

File: vlib/os/os_c.v [hunk 2]
```diff
@@ -177,10 +185,18 @@ pub fn file_size(path string) u64 {
 				println('Using os.file_size() on 32bit systems may not work on big files.')
 			}
 			$if windows {
-				C._wstat(path.to_wide(), voidptr(&s))
+				if C._wstat(path.to_wide(), voidptr(&s)) != 0 {
+					println('os.file_size() Cannot determine file-size: ' +
```
Comment (spytheman) [inline vlib/os/os_c.v:189]: It is better to print errors with `eprintln` so that they go to stderr.

File: vlib/os/os_nix.c.v [hunk 3]
```diff
@@ -47,12 +48,28 @@ mut:
...
+
 fn C.uname(name voidptr) int
 
 fn C.symlink(&char, &char) int
 
 fn C.gethostname(&char, int) int
 
+fn C.getlogin_r(&char, int) int
+
+fn C.utime(&char, voidptr) int
```
Comment (spytheman) [inline vlib/os/os_nix.c.v:64]: Please split the utime addition to another PR. Without it, it would be easier to merge this one.

File: vlib/os/os_nix.c.v (hunk 1 above)
Comment (spytheman) [reply vlib/os/os_nix.c.v:10]: Alex has to prepare a new .zip file with headers in https://github.com/vlang/v/releases/download/0.1.27/linuxroot.zip , that includes utime.h too.

Comment (bstnbuck) [reply vlib/os/os_nix.c.v]: I almost imagined that. The new PR follows soon. Thanks!

File: vlib/os/os_windows.c.v [hunk 4]
```diff
@@ -394,6 +394,11 @@ pub fn hostname() string {
 	return execute('cmd /c hostname').output
 }
 
+pub fn loginname() string {
+	// TODO: use C.GetUserName(&char, u32) bool instead
+	return execute('cmd /c echo %USERNAME%').output
```
Comment (spytheman) [inline vlib/os/os_windows.c.v:399]: hm, would not `os.getenv('USERNAME')` return the same?

Comment (bstnbuck) [reply vlib/os/os_windows.c.v:399]: Yes, you are right. The `os.uname()`, `os.hostname()` and `os.loginname()` functions on Windows are just functional placeholders for now. These should, in my opinion, be replaced by the API functions. The question is, should we do without these API functions and use os.getenv() instead?

Btw currently the API functions always return a Segementation Fault. However, I am working on it.

Comment (bstnbuck) [reply vlib/os/os_windows.c.v:399]: I got this working. But only using TCC and with or without `-autofree`. `-prod` or MSVC fails with C-error or Seg-Fault
```v
fn C.GetComputerNameW(&u16, u32) int

[manualfree]
pub fn hostname() string {
	mut sz := 256 // Optimized length
	mut buf := unsafe { &u16(malloc(256)) }
	defer {
		unsafe { free(buf) }
	}
	if C.GetComputerNameW(voidptr(buf), u32(&sz)) == 0 {
		eprintln(C.errno)
	}
	ret := unsafe { string_from_wide2(buf, sz) }
	return ret
}
```

` C.GetComputerName(&u16, u32) int` does not work, because it needs `Advapi32.dll`.
`os.getenv()` works well.
Are there any suggestions?

File: vlib/os/os_c.v [hunk 5]
```diff
@@ -143,7 +143,7 @@ pub fn truncate(path string, len u64) ? {
 		C.close(fp)
 	}
 	if fp < 0 {
-		return error('open file failed')
+		return error_with_code(posix_get_error_msg(C.errno), C.errno)
```
Comment (dumblob) [inline vlib/os/os_c.v:146]: I hope all these error messages (incl. all below) will not make it to *V 1.0* and instead the proper error will be propagated to the caller - hopefully https://github.com/vlang/v/issues/9367 will solve it.

Comment (bstnbuck) [reply vlib/os/os_c.v:146]: I know it's not the finest way to return errors. However, I think that there here the error is best described. Should the costum type errors also have a place here? The problem is that just opening a file can return all sorts of errors. What do you suggest?

Comment (dumblob) [reply vlib/os/os_c.v:146]: I'm sorry for not being clear - thanks for asking!

It's fine to save the `errno` or maybe directly the `string` (returned by `posix_get_error_msg()`) to the corresponding Error-derived struct and return an `error()` with this struct. But printing out something without letting the user choose whether to print it or not is a bad practice for a library.

Comment (bstnbuck) [reply vlib/os/os_c.v:146]: I now think I know what you mean. 
The user should be able to decide for himself how error messages of various types are handled. However, I prefer to leave the implementation to someone else :)

Comment (dumblob) [reply vlib/os/os_c.v:146]: Sure. I just wanted to link it to the Error tracker issue, so that it can be checked afterwards whether this wasn't skipped/forgotten :wink:.
